src/components/ec-file-upload/ec-file-upload.story.js

```diff
@@ -0,0 +1,56 @@
+import EcFileUpload from './ec-file-upload.vue';
+
+export default {
+  title: 'File Upload',
+  component: EcFileUpload,
+};
+
+export const basic = (args, { argTypes }) => ({
+  components: { EcFileUpload },
+  props: Object.keys(argTypes),
+  data() {
+    return {
+      isFileUploaderDisabled: false,
+      model: [{ name: 'test', type: '.pdf' }],
+    };
+  },
+  methods: {
+    toggleDisable() {
+      this.isFileUploaderDisabled = !this.isFileUploaderDisabled;
+    },
+  },
+  template: `
+    <div class="tw-p-24">
+      <ec-file-upload
+        v-bind="$props"
+        v-model="model"
+        :is-disabled="isFileUploaderDisabled"
+      >
+        <template #title>Drag and drop</template>
+        <template #subtitle>files here or <a
+          href="#"
+          @click.prevent
+          :class="{
+            'tw-cursor-not-allowed' : isFileUploaderDisabled,
+            'tw-text-gray-6' : isFileUploaderDisabled,
+          }">browse</a>.</template>
+        <template #help-text>Attach documentation related to this transaction (ie. invoice(s), customer order, packing list, bill of lading, etc.)</template>
+      </ec-file-upload>
+
+      <button
```

Comment:
We should control this via Controls. Storybook automatically creates a control for each prop including the `is-disabled`. There is no need to have another button in the story for that.

Reply:
In many cases like we have seen previously you do need to have because storybook unfortunately re-renders stuff every time you change a control. In this particular case I wanted to test the scenario

--> Load files
--> Disable
--> Check you cannot delete

But storybook was rerendering and you were loosing the files the moment you were swapping the isDisabled control. 
That said, I introduced in the story a  model where we prepopulate the value so we can test similar scenario that way.

Reply:
I'd rather create another story for that particular scenario.